File: Sesion 11/Marco teorico/3SE LANZA UNA EXCEPCIÓN ( THROW )/3.py
class NegativeValueError(Exception):
    def __init__(self, message):
        self.message = message
        super().__init__(self.message)

def verificar(value):
    if value < 0:
        raise NegativeValueError("Error: Valor negativo no permitido.")


try:
    num = int(input("Ingrese un número: "))
    verificar(num)
    print("El valor es:", num)
except NegativeValueError as e:
    print("Excepción capturada:", e)
except ValueError:
    print("Error: Ingrese un número entero válido.")

File: Sesion 11/Marco teorico/3SE LANZA UNA EXCEPCIÓN ( THROW )/3.cpp
#include <iostream>
#include <exception>
using namespace std;
// Función que lanza una excepción si el valor es negativo
void verificar(int value) {
    if (value < 0) {
        throw runtime_error("Error: Valor negativo no permitido.");
    }
}


int main() {
    int num;


    cout << "Ingrese un número: ";
    cin >> num;


    try {
        verificar(num);
        cout << "El valor es: " << num << endl;
    }
    catch (const exception& e) {
        // Capturamos la excepción y mostramos el mensaje de error
        cerr << "Excepción capturada: " << e.what() << endl;
    }


    return 0;
}

